src/coreComponents/linearAlgebra/interfaces/hypre/HypreExport.cpp

```diff
@@ -193,54 +215,45 @@ void HypreExport::exportVector( HypreVector const & vec,
+    hypre_Vector * wrapperVector{};
     if( MpiWrapper::commRank( vec.getComm() ) == m_targetRank )
     {
       GEOSX_LAI_ASSERT_EQ( values.size(), vec.globalSize() );
       values.move( LvArray::MemorySpace::host, false );
 
       // HACK: create a hypre vector that points to local data; we have to use const_cast,
       //       but this is ok because we don't modify the values, only scatter the vector.
-      localVector = hypre_SeqVectorCreate( LvArray::integerConversion< HYPRE_Int >( values.size() ) );
-      hypre_VectorOwnsData( localVector ) = false;
-      hypre_VectorData( localVector ) = const_cast< real64 * >( values.data() );
-      hypre_SeqVectorInitialize_v2( localVector, HYPRE_MEMORY_HOST );
+      wrapperVector = hypre_SeqVectorCreate( LvArray::integerConversion< HYPRE_Int >( values.size() ) );
+      hypre_VectorOwnsData( wrapperVector ) = false;
+      hypre_VectorData( wrapperVector ) = const_cast< real64 * >( values.data() );
+      hypre_SeqVectorInitialize_v2( wrapperVector, HYPRE_MEMORY_HOST );
     }
 
     // scatter the data
     hypre_ParVector * const parVector = hypre_VectorToParVector( m_subComm,
-                                                                 localVector,
+                                                                 wrapperVector,
                                                                  hypre_ParVectorPartitioning( vec.unwrapped() ) );
     // copy local part of the data over to the output vector
-    HYPRE_Real const * const parVectorData = hypre_VectorData( hypre_ParVectorLocalVector( parVector ) );
-    if( hypre_ParVectorMemoryLocation( vec.unwrapped() ) == HYPRE_MEMORY_HOST )
-    {
-      std::copy( parVectorData, parVectorData + vec.localSize(), vec.extractLocalVector() );
-    }
-    else
-    {
-#ifdef GEOSX_USE_HYPRE_CUDA
-      cudaMemcpy( vec.extractLocalVector(),
-                  parVectorData,
-                  vec.localSize() * sizeof( HYPRE_Real ),
-                  cudaMemcpyHostToDevice );
-#else
-      GEOSX_ERROR( "HypreExport: invalid memory space of target vector" );
-#endif
-    }
+    hypre_TMemcpy( hypre_VectorData( localVector ),
+                   hypre_VectorData( hypre_ParVectorLocalVector( parVector ) ),
+                   HYPRE_Real,
+                   vec.localSize(),
+                   hypre_VectorMemoryLocation( localVector ),
+                   hypre_ParVectorMemoryLocation( parVector ) );
 
     GEOSX_LAI_CHECK_ERROR( hypre_ParVectorDestroy( parVector ) );
-    GEOSX_LAI_CHECK_ERROR( hypre_SeqVectorDestroy( localVector ) );
+    GEOSX_LAI_CHECK_ERROR( hypre_SeqVectorDestroy( wrapperVector ) );
   }
   else
   {
-    hypre_Vector * const localVector = hypre_ParVectorLocalVector( vec.unwrapped() );
-    importArray( hypre_VectorMemoryLocation( localVector ),
+    exportArray( hypre_VectorMemoryLocation( localVector ),
                  values,
                  hypre_VectorData( localVector ) );
   }
+  vec.touch();
```

Comment:
The current `touch()` logic should be explained in detail somewhere (not here).

Reply:
I've documented the usage and intended behavior in doxygen comment for `VectorBase::touch()`, but I don't know where else to put it. It's an implementation detail that is mostly relevant to a group of classes wrapping the same LA package, not external users of these classes.